### PR TITLE
Migrate preference code to AndroidX

### PR DESCRIPTION
**Pull request by maniac103**
Gets rid of deprecation warnings, and ensures consistent UI (e.g. for dialogs) across Android versions.

### CODE REVIEW

**Comment by mueller-ma** [comment]
In xml files there are warnings, that our custom prefs aren't allowed there.
You can remove the icon `ic_none` now. I added this as workaround to have to the items w/o icon indented.

**Comment by maniac103** [comment]
Both done.

File: mobile/src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
-<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.preference.PreferenceScreen
```
Comment (mueller-ma) [inline mobile/src/main/res/xml/preferences.xml]: For consistency I'd like to see this in all pref xml files.

Comment (maniac103) [reply mobile/src/main/res/xml/preferences.xml]: Good point, I'll update that.

**Pull request merged**